JPS_VIRTUALSENSOR/AermodAgent/src/main/java/com/cmclinnovations/aermod/AermodAgent.java

```diff
@@ -169,49 +160,46 @@ public void processRequestParameters(DerivationInputs derivationInputs, Derivati
         }
 
         // create emissions input
-        if (aermod.createPointsFile(ships, srid) != 0) {
+        if (aermod.createPointsFile(allSources, srid) != 0) {
             LOGGER.error("Failed to create points emissions file, terminating");
             return;
         }
 
-        if (!bpi.aermodInputCreated)
-            aermod.createAermodInputFile(scope, nx, ny, srid);
+        aermod.createAermodInputFile(scope, nx, ny, srid);
+        aermod.createAERMODReceptorInput(scope, nx, ny, srid);
         aermod.runAermod("aermod.inp");
-        aermod.runAermod("aermodVirtualSensor.inp");
 
         // Upload files used by scripts within Python Service to file server.
         String outputFileURL = aermod.uploadToFileServer("averageConcentration.dat");
         String outFileURL = aermod.uploadToFileServer("receptor.dat");
-        String sensorFileURL = aermod.uploadToFileServer("virtualSensorConcentrations.dat");
-
-        // Call virtualSensor.py to plot the temporal variation of concentrations
-        JSONArray virtualSensorConcentrations = aermod.plotVirtualSensorData(EnvConfig.PYTHON_SERVICE_SENSOR_URL,
-                sensorFileURL, srid);
-        // Instantiate concentrations at virtual sensor locations as time series
-
-        String startTS = bpi.timeStamps.get(0);
-        LocalDateTime ldr = LocalDateTime.parse(StartTS, DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
-        List<String> timeSteps = bpi.timeStamps;
 
-        queryClient.updateVirtualSensorData(timeSteps, virtualSensorConcentrations);
-
-        List<Double> receptorHeights = bpi.receptorHeights;
+        List<Double> receptorHeights = new ArrayList<>();
+        receptorHeights.add(0.0);
         // Set GeoServer layer names
         List<String> dispLayerNames = new ArrayList<>();
+        String simDir = simulationDirectory.getFileName().toString().replace("-", "");
```

Comment:
what is the purpose of this?

Reply:
The gdalClient doesn't seem to allow hyphens in the dispLayerName string in this line:

Reply:
why do you want to add simDir in the layer name? This can be deduced from the triples markup later on when we do things properly